exchanges/okx/okx_types.go

```diff
@@ -668,27 +661,27 @@ type OrderDetail struct {
 	TradeMode                  string     `json:"tdMode"`
 	TradeID                    string     `json:"tradeId"`
 	FillTime                   time.Time  `json:"fillTime"`
-	Source                     string     `json:"source,omitempty"`
+	Source                     string     `json:"source"`
 	State                      string     `json:"state"`
 	TakeProfitTriggerPriceType string     `json:"tpTriggerPxType"`
 	StopLossTriggerPriceType   string     `json:"slTriggerPxType"`
 	StopLossOrdPx              string     `json:"slOrdPx"`
 	RebateCurrency             string     `json:"rebateCcy"`
 	QuantityType               string     `json:"tgtCcy"`   // base_ccy and quote_ccy
 	Category                   string     `json:"category"` // normal, twap, adl, full_liquidation, partial_liquidation, delivery, ddh
-	AccumulatedFillSize        float64    `json:"accFillSz,string"`
-	FillPrice                  float64    `json:"fillPx,string"`
-	FillSize                   float64    `json:"fillSz,string"`
+	AccumulatedFillSize        float64    `json:"accFillSz"`
+	FillPrice                  float64    `json:"fillPx"`
+	FillSize                   float64    `json:"fillSz"`
 	RebateAmount               float64    `json:"rebate"`
 	FeeCurrency                string     `json:"feeCcy"`
-	TransactionFee             float64    `json:"fee,string"`
-	AveragePrice               float64    `json:"avgPx,string"`
-	Leverage                   float64    `json:"lever,string"`
-	Price                      float64    `json:"px,string"`
-	Size                       float64    `json:"sz,string"`
-	TakeProfitTriggerPrice     float64    `json:"tpTriggerPx,string"`
-	TakeProfitOrderPrice       float64    `json:"tpOrdPx,string"`
-	StopLossTriggerPrice       float64    `json:"slTriggerPx,string"`
+	TransactionFee             float64    `json:"fee"`
+	AveragePrice               float64    `json:"avgPx"`
+	Leverage                   float64    `json:"lever"`
+	Price                      float64    `json:"px"`
+	Size                       float64    `json:"sz"`
+	TakeProfitTriggerPrice     float64    `json:"tpTriggerPx"`
+	TakeProfitOrderPrice       float64    `json:"tpOrdPx"`
+	StopLossTriggerPrice       float64    `json:"slTriggerPx"`
```

Comment:
These changes do not look to work with the API guidelines: https://www.okx.com/docs-v5/en/#rest-api-trade-get-order-details
Did you place an order and get an error?

Reply:
There is a json convert method in the okx_convert.go to unmarshal those fields